todo/todo.py

```diff
@@ -183,344 +173,370 @@ async def showsettings(self, ctx):
+        indexes = [x - 1 for x in indexes]
+        _ = await self.config.user(ctx.author).completed()
+        if not _:
+            return await ctx.send(_no_completed_message.format(prefix=ctx.clean_prefix))
+        else:
+            removed, failed = 0, 0
+            async with ctx.typing():
+                async with self.config.user(ctx.author).completed() as comp:
+                    for index in indexes:
+                        try:
+                            comp.pop(index)
+                        except IndexError:
+                            failed += 1
+                        else:
+                            removed += 1
+            msg = "Done."
+            if await self.config.user(ctx.author).detailed_pop():
+                if removed:
+                    msg += f"\nRemoved: {removed} {'todos' if removed > 1 else 'todo'}"
+                if failed:
+                    msg += f"\nFailed to remove: {failed} {'todos' if failed > 1 else 'todo'}"
+            await ctx.send(msg)
+            await self._maybe_autosort(ctx)
+
+    @complete.command(name="removeall", aliases=["delall", "rma"])
+    async def complete_removeall(self, ctx):
+        """Remove all of your completed todos"""
+        await ctx.send(
+            "WARNING, this will remove **ALL** of your completed todos. Would you like to remove them? (y/n)"
+        )
+        pred = MessagePredicate.yes_or_no(ctx)
         try:
-            todo = await self.config.user(ctx.author).todos()[act_index]
-        except IndexError:
-            return await ctx.send("Hm, that doesn't seem to be a todo!")
-
-        todo = f"{index}. {todo}"
-        md = await self.config.user(ctx.author).use_md()
-        embedded = await self.config.user(ctx.author).use_embeds()
-
-        if md:
-            todo = box(todo, "md")
-        if embedded:
-            sending = discord.Embed(
-                title=f"{ctx.author.name}'s ToDos",
-                description=todo,
-                colour=await ctx.embed_colour(),
-            )
-            sending.set_footer(text="ToDo list")
-            key = "embed"
+            await self.bot.wait_for("message", check=pred)
+        except asyncio.TimeoutError:
+            await ctx.send("Okay, I won't delete your completed todos")
         else:
-            sending = todo
-            key = "content"
-        channel = await self.get_destination(ctx)
-        await channel.send(**{key: sending})
-
-    @todo.command()
-    async def explain(self, ctx, comic: bool = False):
-        """Explain what a todo is and why this cog exists!"""
-        if comic:
-            if await ctx.embed_requested():
-                embed = discord.Embed(
-                    title="About ToDo", colour=await ctx.embed_colour()
-                )
-                embed.set_footer(text="Jojo's ToDo Cog")
-                embed.set_image(url=_comic_link)
-                kwargs = {"embed": embed}
+            if pred.result is False:
```

Comment:
We can simplify this as `result` will only return a bool.

```suggestion
            if not pred.result:
```